### ISSUE TITLE
Works with partition table snippets?

### ISSUE THREAD

**Issue by cacack** [theforeman/foreman_templates#24]
We're using Foreman 1.10.3 and have a number of partition table snippets.  I'd like to use this plugin to migrate to a git repository, but the plugin doesn't appear to support using snippets in partition tables.  Code leads me to believe snippets will be imported as provisioning templates.  Is this correct?

**Comment by GregSutcliffe** [comment]
It shouldn't matter how you structure it. PTable templates should be correctly imported (see https://github.com/theforeman/foreman_templates/blob/master/lib/templates.rake#L81-L106 for the handler on ptable entries) and snippets (which are a different thing - they are only text with no associations or types) will be imported using this 9https://github.com/theforeman/foreman_templates/blob/master/lib/templates.rake#L108-L126)

**Comment by cacack** [comment]
Thanks for the quick response!

In Foreman, we have some partition table templates with snippet checked and some not checked.  But it appears that doesn't matter to Foreman as you can use the snippet macro to source in any template of like type, not just ones marked as snippet.  I just happened to sample the ones that were marked as snippets leading to my assumption.

**Comment by cacack** [comment]
I did a little more digging and discovered that Foreman supports snippets that are stored with [provision templates](https://theforeman.org/api/1.12/apidoc/v2/provisioning_templates/create.html) AND [partition tables](https://theforeman.org/api/1.12/apidoc/v2/ptables/create.html) by using the snippet param.  Foreman doesn't treat snippets as a separate [kind type](https://theforeman.org/api/1.12/apidoc/v2/template_kinds/index.html).

Can the plugin be modified to mirror this behavior?

I think the template metadata would need to support a snippet flag when the type is provision or ptable. Something like:
```
<%#
kind: ptable
snippet: true
name: my_partition_table
...
%>
```

**Comment by GregSutcliffe** [comment]
Fixed in 11ee3cb79d00dfd82b44d8ec70c87772853b1715
